src/action-menu/_ActionUtils.jsx

```diff
@@ -0,0 +1,193 @@
+import React from 'react';
+import {
+  KEY_RETURN,
+  KEY_SPACE,
+  KEY_RIGHT,
+  KEY_LEFT,
+  KEY_UP,
+  KEY_DOWN,
+  KEY_HOME,
+  KEY_END,
+} from 'keycode-js';
+
+/**
+ * Enables focus styles for the target of the given event. Typically used as an onBlur callback on selectable elements.
+ */
+const enableFocusStyles = (event) => {
+  event.currentTarget.setAttribute('data-focus-styles-enabled', 'true');
+};
+
+/**
+ * Disables focus styles for the target of the given event. Typically used as an onMouseDown callback on selectable elements.
+ */
+const disableFocusStyles = (event) => {
+  event.currentTarget.setAttribute('data-focus-styles-enabled', 'false');
+};
+
+/**
+ * Validates whether or not the char is acceptable for use in a first char search,
+ * @param {string} char The char to validate.
+ */
+const isValidChar = char => char.length === 1 && char.match(/\S/);
+
+/**
+ * Returns a function that will execute the provided function upon detection of a KEY_RETURN or KEY_SPACE keydown event.
+ * @param {string} key The uniquely identifiable key.
+ * @param {Function} onAction The function to be executed after event detection.
+ * @param {Function} onArrow The function to be executed after event detection.
+ * @param {Function} onChar The function to be executed after event detection.
+ */
+const generateOnKeyDown = (key, onAction, onArrow, onChar) => (
+  event => {
+    if (event.altKey || event.ctrlKey || event.metaKey) {
+      return;
+    }
+
+    let shouldPrevent = false;
+    if (event.nativeEvent.keyCode === KEY_RETURN || event.nativeEvent.keyCode === KEY_SPACE) {
+      shouldPrevent = true;
```

Comment:
is it worth turning our preventDefault/stopPropagation logic into its own function, and invoking that, instead of invoking at the end based on a boolean? might not be worth the squeeze, i just think this is a little confusing haha